ci/Dockerfile.ubuntu-with-grpc

```diff
@@ -0,0 +1,36 @@
+# Copyright 2017 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+ARG DISTRO=ubuntu
+ARG DISTRO_VERSION=16.04
+FROM cached-${DISTRO}-${DISTRO_VERSION}:tip AS staging
+MAINTAINER "Carlos O'Ryan <coryan@google.com>"
+
+RUN apt-get install -y automake libtool build-essential
+
+ARG NCPU=2
+ARG CXX=g++
```

Comment:
If you care about speed of compilation, I believe Clang should be faster in compiling than GCC, if that's on your critical path.

Reply:
Done.
Thought these are the defaults and the `.travis.yml` file changed them to `clang++` and `clang`.